src/io/input/xml/XMLFileReader.cpp

```diff
@@ -3,26 +3,85 @@
 #include <optional>
 
 #include "io/logger/Logger.h"
+#include "io/particle_spawners/CuboidSpawner.h"
 
-SimulationParams XMLFileReader::readConfiguration(const std::string& filepath, ParticleContainer& particle_container) const {
-    if (filepath.empty()) {
-        Logger::logger->error("No input file specified.");
-        exit(-1);
-    }
-
+SimulationParams XMLFileReader::readFile(const std::string& filepath, ParticleContainer& particle_container) const {
     try {
         std::unique_ptr< ::configuration> config = configuration_(filepath);
 
-        Logger::logger->info("Successfully parsed XML file.");
+        Logger::logger->info("Successfully parsed XML file.Simulation parameters: ");
         Logger::logger->info("FPS: {}", config->fps().get());
+        Logger::logger->info("Video length: {}", config->video_length().get());
+        Logger::logger->info("End time: {}", config->end_time().get());
+        Logger::logger->info("Delta t: {}", config->delta_t().get());
+
+        if(config->fps().get() < 0){
+            Logger::logger->error("FPS must be positive");
```

Comment:
Since this is identical to the CLI parsing, the "range checks" can be combined in a single place.

maybe put it in the `SimulationParams` class to avoid duplicated logic

Reply:
Maybe the default values can be stored in this class too. I dont know if its possible to leave out XML entries to fall back on default values